src/Redistribution.sol

```diff
@@ -328,45 +329,39 @@ contract Redistribution is AccessControl, Pausable {
         }
 
         bytes32 commitHash = wrapCommit(_overlay, _depth, _hash, _revealNonce);
+        uint256 id = findCommit(_overlay, commitHash, currentCommits.length);
 
-        uint256 commitsArrayLength = currentCommits.length;
-
-        for (uint256 i = 0; i < commitsArrayLength; i++) {
-            if (currentCommits[i].overlay == _overlay && commitHash == currentCommits[i].obfuscatedHash) {
-                require(
-                    inProximity(currentCommits[i].overlay, currentRevealRoundAnchor, _depth),
-                    "anchor out of self reported depth"
-                );
-                //check can only revealed once
-                require(currentCommits[i].revealed == false, "participant already revealed");
-                currentCommits[i].revealed = true;
-                currentCommits[i].revealIndex = currentReveals.length;
-
-                currentReveals.push(
-                    Reveal({
-                        owner: currentCommits[i].owner,
-                        overlay: currentCommits[i].overlay,
-                        stake: currentCommits[i].stake,
-                        stakeDensity: currentCommits[i].stake * uint256(2 ** _depth),
-                        hash: _hash,
-                        depth: _depth
-                    })
-                );
-
-                emit Revealed(
-                    cr,
-                    currentCommits[i].overlay,
-                    currentCommits[i].stake,
-                    currentCommits[i].stake * uint256(2 ** _depth),
-                    _hash,
-                    _depth
-                );
-
-                return;
-            }
-        }
+        // Check that the commit exists,
+        require(id != type(uint256).max, "no matching commit or hash");
```

Comment:
IMO it would be nicer to throw an exception in the `findCommit` function instead, because this check requires to know that the findCommit function will set `id` to `type(uint256).max` if there is no match.